### ISSUE TITLE
Frontend (angular-ui-router): State change events are deprecated

### ISSUE THREAD

**Issue by spyshiv** [Cloud-CV/EvalAI#1272]
**Issue:** 
In `angular-ui-router<=1.x`,  State change events are deprecated, DISABLED and replaced by Transition Hooks. So redirection and state change will not work if lnstalled version of `angular-ui-router` is `>=1.x`. 
[More info](https://github.com/angular-ui/ui-router/wiki#state-change-events)

**How to Fix:**
* change bower dependencies `"angular-ui-router": "^0.3.1"` to  `"angular-ui-router": "0.3.1"` (without `^`) so that installation will be restricted to version `0.3.1`
or 
* if we need to go with latest version of angular-ui-router then we have to replace `$rootScope.$on('$stateChangeStart', function(){ ... })` with `$transitions.onStart({...})` and also have to update bower dependency for the same.

@aka-jain @RishabhJain2018 @deshraj

**Comment by xamfy** [comment]
I want to do this, should I just update the dependencies?

**Comment by RishabhJain2018** [comment]
@xamfy I think you are currently working on #1247 . Please complete that first then you can take this up as it blocks the development of the whole project. Till then it is open for everyone. :)

**Comment by Ayukha** [comment]
Can i try ? @RishabhJain2018

**Comment by RishabhJain2018** [comment]
@Ayukha Yes you can.

**Comment by RishabhJain2018** [comment]
@spyshiv @deshraj @aka-jain I would prefer to go with the second approach of @spyshiv as in future many dependency conflicts can  be avoided by using the latest version of `angular-ui-router`. What do you say?

**Comment by spyshiv** [comment]
second approach 👍

**Comment by sblcook** [comment]
I can give the second one a shot if @Ayukha isn't working on it. I only found three uses of the old method when I searched.

**Comment by aka-jain** [comment]
Second aproach doesn't seem like a good option for now as we are using state related functions heavily in project and updation of ui router might break functionality. We can only continue with the second approach once we are confident with the change. 
Let's stick to the same version or first list all the functions that has been depricated in current version of UI ROUTER.

**Comment by aayusharora** [comment]
Not sure @spyshiv @aka-jain If we are planning long-term goals here, removing `^` is the solution for temporary purposes, I am in doubt that we need to maintain different packages like them too and have to restrict ourselves to the old AngularJS.

**Comment by aka-jain** [comment]
I also agree to go with the new version but first we have to be confident about all the changes that have been made to new update. 
I've already tried out the latest version of UI router and there are lots of changes. We should only continue with the new version when all the changes and usage get listed down otherwise chances of code breakage will become high.

Let's do a small POC on this and will check wether all functionalities work fine or not with the new update. Till then let's stick to the older version.
@aayusharora

**Comment by aayusharora** [comment]
Agree @aka-jain, which UI router version have you tried ?

**Comment by spyshiv** [comment]
@aka-jain @RishabhJain2018 I am assigning this to me.

**Comment by Obtin** [comment]
@spyshiv, Is this issue resolved? If not I would like to try this.

**Comment by spyshiv** [comment]
@Obtin : Sure!! Go ahead. 
Assigning this to you.
Let me know if you face any issue or need any guidence regarding this.

**Comment by Obtin** [comment]
@spyshiv, please correct me if I am wrong, according to the above discussion the second approach is to be followed?

**Comment by spyshiv** [comment]
You're right. Second approach.
